mcu-util/src/orb/main_board.rs

```diff
@@ -328,9 +328,10 @@ impl MainBoardInfo {
     }
 
     /// Fetches `MainBoardInfo` from the main board
+    /// doesn't fail, but lazily fetches as much info as it could
     /// on timeout, returns the info that was fetched so far
```

Comment:
At first, I was about to comment asking why would we have a timeout in async code since the caller can just use `tokio::time::timeout()`. Then I investigated further, and noticed that we call blocking code from this async function. I thought we removed all of that 😭

I'll open a separate PR to fix this, since its unrelated to this PR. It is *extremely important* not to block in async code, as it leads to very hard to debug behavior and is often a cause for subtle bugs.

Reply:
https://linear.app/worldcoin/issue/O-2706/blocking-in-async-in-main-board-and-sec-board